[4196] Filter studies by AWS region

Luke-Sikina opened the pull request:
Description of changes:
- Pull AWS region from yaml during scrape
- Add region to db
- Filter results for `list` in study service

srpiatt commented on addons/addon-base-raas/packages/base-raas-services/lib/schema/create-study.json:
```diff
@@ -38,6 +42,12 @@
+    "region": {
+      "type": "string",
+      "maxLength": 64,
+      "default": "unknown",
```
You could try adding a validation pattern.

Some variations on [https://www.regextester.com/109163](https://www.regextester.com/109163):
- `(us(-gov)?|af|ap|ca|cn|eu|me|sa)-(central|(north|south)?(east|west)?)-\d+[a-z]?` expands allowed country list and adds az letter, matches `us-west-2` or `af-east-1a`.
- `(us(-gov)?|[a-z]{2})-(central|(north|south)?(east|west)?)-\d+[a-z]?` Same as above, but generalizes the country letters for the future

Luke-Sikina replied:
I don't think this is a good idea. While validating the data might be attractive, we aren't the authority on what an AWS region will be. Without being able to tie the regex to a published, maintained standard, I think this would be a net negative.

Luke-Sikina replied:
TODO: check to see if this gets sanitized

Luke-Sikina replied:
TODONE: it does. The DynamoDB rest api handles this.

srpiatt commented on addons/addon-base-raas/packages/base-raas-services/lib/study/study-service.js:
```diff
@@ -518,7 +518,11 @@ class StudyService extends Service {
+          // filter by aws region. Because the scraper might not run until after
+          // this code is deployed we want this filter to be permissive when it isn't
+          // clear if the study is in the correct region
+          .filter(study => ['us-east-1', 'unknown', undefined].includes(study.region))
```
I bet there's a place where we can pass in the region we're in (env vars?), instead of hard-coding the `us-east-1` region we want.

Luke-Sikina replied:
Good idea.

Luke-Sikina committed:
[4196] Filter studies by AWS region

- Scrape script: add region to json
- Sample service:
  - filter db results by region
  - make region env var
- Misc: fix a test I broke in another PR (sorry)

srpiatt reviewed (approved):
Contingent on verifying that data is sanitized before it gets to the db. :)
